Assets/Scripts/CharacterStats.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace ARPGTD.CharacterStats{
+    /// CharacterStat is a class built to handle changes in character stats
+    /// The stats themselves will be stored in the Character object
+    [Serializable]
+    public class CharacterStat
+    {
+        public float BaseValue;                             //Base value of the stat
+
+        public virtual float Value {                                //The value of the stat during final value calculation
+            get {
+                if (isDirty || BaseValue != lastBaseValue)
+                {
+                    lastBaseValue = BaseValue;
+                    _value = CalculateFinalValue();
+                    isDirty = false;
+                }
+                return CalculateFinalValue();
+            }
+        }
+
+        protected bool isDirty = true;                        //Boolean to describe if a change has been made
+        protected float _value;                               //Last known value of the stat - set in Value
+        protected float lastBaseValue = float.MinValue;       //Last known base value of the stat
+
+        protected readonly List<StatModifier> statModifiers;
+        public readonly ReadOnlyCollection<StatModifier> StatModifiers;
+
+        /// Creator
+        public CharacterStat()
+        {
+            statModifiers = new List<StatModifier>();
+            StatModifiers = statModifiers.AsReadOnly();
+        }
+
+        ///Creator
+        public CharacterStat(float baseValue) : this()
+        {
+            BaseValue = baseValue;
+        }
+
+        ///Compare the order for sorting, making the multiplication happen after addition to the stats
+        protected virtual int CompareModifierOrder(StatModifier a, StatModifier b)
+        {
+            if (a.Order < b.Order)
```

Comment:
Wrap these returns in brackets, will make adding to this easier. Not to mention it will separate the statements a bit more as well, they seem to blend in with each other on first glance